File: algorithm/01-two-sum/twoSum.py
# 一刷时间: 2019
# 链接：https://leetcode-cn.com/problems/two-sum

class Solution:
    def twoSum(self, nums, target):
        hashmap = {}
        for i, val in enumerate(nums):
            complement = target - val
            if complement in hashmap:
                return [hashmap[complement], i]
            hashmap[val] = i
        return []

    def __init__(self, nums, target):
        print(self.twoSum(nums, target))


nums = [2, 7, 11, 15]
target = 9
Solution(nums, target)

File: algorithm/167-Two-Sum-II-Input-array-is-sorted/two-sum.js
// 一刷时间: 2020-04
// 来源: https://leetcode-cn.com/problems/two-sum-ii-input-array-is-sorted/

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let map = {};
  const len = numbers.length;
  for (let i = 0; i < len; i++) {
    const item = numbers[i];
    const complement = target - item;
    const curItem = map[complement] + 1;
    if (curItem) {
      return [curItem, i + 1];
    }
    map[item] = i;
  }
  return [];
};
module.exports = twoSum;
